### PR TITLE
Update to use FMS 2023.04

### PR DESCRIPTION
**Pull request by shlyaeva**
All work done by @climbfuji and @travissluka, I'm just raking up PR points.

jedi-ci-bundle-branch=feature/no_gfdl_forks
jedi-ci-next=true

build-group=https://github.com/JCSDA-internal/fv3-jedi-linearmodel/pull/17
build-group=https://github.com/JCSDA/GFDL_atmos_cubed_sphere/pull/6
build-group=https://github.com/JCSDA-internal/fv3-jedi/pull/812

### CODE REVIEW

**Comment by climbfuji** [comment]
I don't see any CI runs?

**Comment by shlyaeva** [comment]
I just made it non-draft a few minutes ago; we'll need to trigger CI.

**Comment by eap** [comment]
I temporarily patched a change to the CI manifest that should allow this to build. It has already gotten past the first configuration step so we should have the results soon.

I've already reverted the change (since it was global) so additional pushes will fail quickly as before.

File: CMakeLists.txt [hunk 1]
```diff
@@ -47,10 +48,7 @@ include_directories( ${NETCDF_INCLUDE_DIRS} )
 # Model
 # TODO update to allow switching between mom6solo (here) and UFS (external)
```
Comment (travissluka) [inline CMakeLists.txt:49]: note to self, or anyone else that cares, this is fine for now but I'm assuming we're building MOM6 twice when soca is built as part of the ufs-bundle. No real harm, but wasteful. At some point special logic should be added to use a prebuilt mom6 lib if it exists.

**Review by travissluka — APPROVED** [review]
I see the modified CI worked.
Builds and tests pass on my machine (with develop merged in)

It'll be nice to remove the JCSDA FMS and MOM6 repo forks !

**Comment by svahl991** [comment]
Just a reminder that the coordinated merge for this is planned for Monday, July 1. Looks like this still needs another approval.

**Comment by svahl991** [comment]
@HamidehGMAO I added you to the reviewers on this since it appears to need a little maintenance, and Travis is on PTO, and this is scheduled to be merged on Monday.

**Comment by travissluka** [comment]
@kbhargava @HamidehGMAO @guillaumevernieres could one of you review this this morning? it's supposed to be merged today. The CI tests are expected to fail (the CI had to be temporarily modified earlier for it to successfully pass), but they did pass earlier, and building outside of CI does work correctly

**Comment by HamidehGMAO** [comment]
@travissluka @svahl991 I am looking at this right now.

**Comment by HamidehGMAO** [comment]
Which spack stack versions is this supposed to work with? I can't build it with spack spack 1.5.1 gnu on discover, I'm going to test it with a newer version. 
image

**Comment by HamidehGMAO** [comment]
I tried with spack-stack 1.7.0 and still it failed, 
image

**Comment by kbhargava** [comment]
> @kbhargava @HamidehGMAO @guillaumevernieres could one of you review this this morning? it's supposed to be merged today. The CI tests are expected to fail (the CI had to be temporarily modified earlier for it to successfully pass), but they did pass earlier, and building outside of CI does work correctly

Let me test on hercules/orion

**Comment by shlyaeva** [comment]
See https://github.com/orgs/JCSDA-internal/discussions/139: you'll need to unload fms and load a different version.

**Comment by HamidehGMAO** [comment]
> See https://github.com/orgs/JCSDA-internal/discussions/139: you'll need to unload fms and load a different version.

Ah! I see! That should solve this error. Thanks Anna!

**Comment by HamidehGMAO** [comment]
hmm, now I get this error. Am I missing something?
image

**Review by guillaumevernieres — APPROVED** [review]

**Comment by kbhargava** [comment]
Just an update I am having issues on both hercules and orion when I run `ecbuild ../bundle`. I get the following errors

```
-- Configuring done
CMake Error at soca/mom6/CMakeLists.txt:28 (add_library):
  Cannot find source file:

    MOM6/src/ALE/MOM_ALE.F90

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .ixx .cppm .h
  .hh .h++ .hm .hpp .hxx .in .txx .f .F .for .f77 .f90 .f95 .f03 .hip .ispc


CMake Error at soca/mom6/CMakeLists.txt:48 (add_executable):
  Cannot find source file:

    MOM6/config_src/drivers/solo_driver/MESO_surface_forcing.F90

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .ixx .cppm .h
  .hh .h++ .hm .hpp .hxx .in .txx .f .F .for .f77 .f90 .f95 .f03 .hip .ispc


CMake Error at soca/mom6/CMakeLists.txt:28 (add_library):
  No SOURCES given to target: mom6


CMake Error at soca/mom6/CMakeLists.txt:48 (add_executable):
  No SOURCES given to target: mom6solo


CMake Generate step failed.  Build files cannot be regenerated correctly.
```

**Comment by climbfuji** [comment]
> hmm, now I get this error. Am I missing something? image

that's a known issue, you need to load the sp module

**Comment by climbfuji** [comment]
> Just an update I am having issues on both hercules and orion when I run `ecbuild ../bundle`. I get the following errors
> 
> ```
> -- Configuring done
> CMake Error at soca/mom6/CMakeLists.txt:28 (add_library):
>   Cannot find source file:
> 
>     MOM6/src/ALE/MOM_ALE.F90
> 
>   Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .ixx .cppm .h
>   .hh .h++ .hm .hpp .hxx .in .txx .f .F .for .f77 .f90 .f95 .f03 .hip .ispc
> 
> 
> CMake Error at soca/mom6/CMakeLists.txt:48 (add_executable):
>   Cannot find source file:
> 
>     MOM6/config_src/drivers/solo_driver/MESO_surface_forcing.F90
> 
>   Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .ixx .cppm .h
>   .hh .h++ .hm .hpp .hxx .in .txx .f .F .for .f77 .f90 .f95 .f03 .hip .ispc
> 
> 
> CMake Error at soca/mom6/CMakeLists.txt:28 (add_library):
>   No SOURCES given to target: mom6
> 
> 
> CMake Error at soca/mom6/CMakeLists.txt:48 (add_executable):
>   No SOURCES given to target: mom6solo
> 
> 
> CMake Generate step failed.  Build files cannot be regenerated correctly.
> ```

recursive checkout? is the file there?

**Comment by travissluka** [comment]
> Just an update I am having issues on both hercules and orion when I run `ecbuild ../bundle`. I get the following errors
> 
> ```
> -- Configuring done
> CMake Error at soca/mom6/CMakeLists.txt:28 (add_library):
>   Cannot find source file:
> 
>     MOM6/src/ALE/MOM_ALE.F90
> 
>   Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .ixx .cppm .h
>   .hh .h++ .hm .hpp .hxx .in .txx .f .F .for .f77 .f90 .f95 .f03 .hip .ispc
> 
> 
> CMake Error at soca/mom6/CMakeLists.txt:48 (add_executable):
>   Cannot find source file:
> 
>     MOM6/config_src/drivers/solo_driver/MESO_surface_forcing.F90
> 
>   Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .ixx .cppm .h
>   .hh .h++ .hm .hpp .hxx .in .txx .f .F .for .f77 .f90 .f95 .f03 .hip .ispc
> 
> 
> CMake Error at soca/mom6/CMakeLists.txt:28 (add_library):
>   No SOURCES given to target: mom6
> 
> 
> CMake Error at soca/mom6/CMakeLists.txt:48 (add_executable):
>   No SOURCES given to target: mom6solo
> 
> 
> CMake Generate step failed.  Build files cannot be regenerated correctly.
> ```

@kbhargava @HamidehGMAO  if you're building as part of some other bundle (ie jedi or ufs) it should be fine. however if you use the soca bundle, you'll need to make sure you did a recursive checkout of soca, from within the soca source directory:

```
 git submodule update --init --recursive 
```

**Comment by kbhargava** [comment]
> > Just an update I am having issues on both hercules and orion when I run `ecbuild ../bundle`. I get the following errors
> > ```
> > -- Configuring done
> > CMake Error at soca/mom6/CMakeLists.txt:28 (add_library):
> >   Cannot find source file:
> > 
> >     MOM6/src/ALE/MOM_ALE.F90
> > 
> >   Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .ixx .cppm .h
> >   .hh .h++ .hm .hpp .hxx .in .txx .f .F .for .f77 .f90 .f95 .f03 .hip .ispc
> > 
> > 
> > CMake Error at soca/mom6/CMakeLists.txt:48 (add_executable):
> >   Cannot find source file:
> > 
> >     MOM6/config_src/drivers/solo_driver/MESO_surface_forcing.F90
> > 
> >   Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .ixx .cppm .h
> >   .hh .h++ .hm .hpp .hxx .in .txx .f .F .for .f77 .f90 .f95 .f03 .hip .ispc
> > 
> > 
> > CMake Error at soca/mom6/CMakeLists.txt:28 (add_library):
> >   No SOURCES given to target: mom6
> > 
> > 
> > CMake Error at soca/mom6/CMakeLists.txt:48 (add_executable):
> >   No SOURCES given to target: mom6solo
> > 
> > 
> > CMake Generate step failed.  Build files cannot be regenerated correctly.
> > ```
> 
> @kbhargava @HamidehGMAO if you're building as part of some other bundle (ie jedi or ufs) it should be fine. however if you use the soca bundle, you'll need to make sure you did a recursive checkout of soca, from within the soca source directory:
> 
> ```
>  git submodule update --init --recursive 
> ```

works now. Always tend to forget recursive stuff. Thanks @travissluka and @climbfuji

**Comment by HamidehGMAO** [comment]
Thanks @climbfuji , I loaded the sp modules, and it works fine now. Everything looks good now! Thanks.

**Review by HamidehGMAO — APPROVED** [review]

**Review by kbhargava — APPROVED** [review]
Builds fine and ctests pass on orion. 🎉

**Pull request merged**